### PR TITLE
Use an autoformatter for the indentation.

### PR DESCRIPTION
**Pull request by realityking**

### CODE REVIEW

File: libraries/joomla/cache/controller/view.php [hunk 1]
```diff
@@ -69,7 +69,7 @@ class JCacheControllerView extends JCacheController
 			}
 
 			else
-			{ // No workarounds, so all data is stored in one piece
+				{ // No workarounds, so all data is stored in one piece
 				echo (isset($data)) ? $data : null;
```
Comment (elkuku) [inline libraries/joomla/cache/controller/view.php:73]: The formatter failed :P - put the comment on a new line ?

File: libraries/joomla/cache/storage/file.php [hunk 2]
```diff
@@ -197,10 +197,10 @@ class JCacheStorageFile extends JCacheStorage
...
 				{
-					if ($folders[$i] != $folder)
-					{
-						$return |= $this->_deleteFolder($this->_root . '/' . $folders[$i]);
-					}
+						if ($folders[$i] != $folder)
+						{
+							$return |= $this->_deleteFolder($this->_root . '/' . $folders[$i]);
+						}
 				}
```
Comment (elkuku) [inline libraries/joomla/cache/storage/file.php:204]: fail

File: libraries/joomla/filesystem/file.php [hunk 3]
```diff
@@ -492,7 +492,7 @@ class JFile
 			else
 			{
 				if (is_writeable($baseDir) && move_uploaded_file($src, $dest))
-				{ // Short circuit to prevent file permission errors
+					{ // Short circuit to prevent file permission errors
 					if (JPath::setPermissions($dest))
```
Comment (elkuku) [inline libraries/joomla/filesystem/file.php:497]: fail

File: libraries/joomla/filesystem/stream.php [hunk 4]
```diff
@@ -237,17 +237,17 @@ class JStream extends JObject
...
 
-			// bzip2 is much like gzip except it doesn't use the include path
+				// bzip2 is much like gzip except it doesn't use the include path
 			case 'bz':
 				$this->_fh = bzopen($filename, $mode);
 				break;
 
-			// fopen can handle streams
+				// fopen can handle streams
 			case 'f':
```
Comment (elkuku) [inline libraries/joomla/filesystem/stream.php:251]: those 3 look strange..

File: libraries/joomla/form/fields/groupedlist.php [hunk 5]
```diff
@@ -117,7 +117,7 @@ class JFormFieldGroupedList extends JFormField
 					}
 					break;
 
-				// Unknown element type.
+					// Unknown element type.
 				default:
```
Comment (elkuku) [inline libraries/joomla/form/fields/groupedlist.php:121]: 5 fails

File: libraries/joomla/form/form.php [hunk 6]
```diff
@@ -1311,7 +1311,7 @@ class JForm
 
 				break;
 			default:
-			// Check for a callback filter.
+				// Check for a callback filter.
 				if (strpos($filter, '::') !== false && is_callable(explode('::', $filter)))
```
Comment (elkuku) [inline libraries/joomla/form/form.php:1315]: 9 fails

File: libraries/joomla/form/rules/url.php [hunk 7]
```diff
@@ -75,7 +75,7 @@ class JFormRuleUrl extends JFormRule
 		}
 		// For some schemes here must be two slashes.
 		if (($urlScheme == 'http' || $urlScheme == 'https' || $urlScheme == 'ftp' || $urlScheme == 'sftp' || $urlScheme == 'gopher'
-			|| $urlScheme == 'wais' || $urlScheme == 'gopher' || $urlScheme == 'prospero' || $urlScheme == 'telnet')
+				|| $urlScheme == 'wais' || $urlScheme == 'gopher' || $urlScheme == 'prospero' || $urlScheme == 'telnet')
 			&& ((substr($value, strlen($urlScheme), 3)) !== '://'))
```
Comment (elkuku) [inline libraries/joomla/form/rules/url.php:79]: fail

File: libraries/joomla/installer/adapters/template.php [hunk 8]
```diff
@@ -131,7 +131,7 @@ class JInstallerTemplate extends JAdapterInstance
 				$this->parent->setOverwrite(true);
 				$this->parent->setUpgrade(true);
 				if ($id)
-				{ // if there is a matching extension mark this as an update; semantics really
+					{ // if there is a matching extension mark this as an update; semantics really
 					$this->route = 'update';
```
Comment (elkuku) [inline libraries/joomla/installer/adapters/template.php:136]: fail

File: libraries/joomla/string/string.php [hunk 9]
```diff
@@ -819,7 +819,7 @@ abstract class JString
 							|| (4 < $mBytes)
 							|| (($mUcs4 & 0xFFFFF800) == 0xD800) // From Unicode 3.2, surrogate characters are illegal
 							|| ($mUcs4 > 0x10FFFF)) // Codepoints outside the Unicode range are illegal
-						{
+							{
 							return false;
```
Comment (elkuku) [inline libraries/joomla/string/string.php:823]: fail

File: libraries/joomla/updater/adapters/extension.php [hunk 10]
```diff
@@ -44,7 +44,7 @@ class JUpdaterExtension extends JUpdateAdapter
 				$this->current_update->update_site_id = $this->_update_site_id;
 				$this->current_update->detailsurl = $this->_url;
 				break;
-			// Don't do anything
+				// Don't do anything
 			case 'UPDATES':
```
Comment (elkuku) [inline libraries/joomla/updater/adapters/extension.php:48]: fail

File: libraries/joomla/updater/adapters/extension.php [hunk 11]
```diff
@@ -100,7 +100,7 @@ class JUpdaterExtension extends JUpdateAdapter
 				}
 				break;
 			case 'UPDATES':
-			// :D
+				// :D
 				break;
```
Comment (elkuku) [inline libraries/joomla/updater/adapters/extension.php:104]: right :D

File: libraries/joomla/updater/update.php [hunk 12]
```diff
@@ -228,7 +228,7 @@ class JUpdate extends JObject
 				}
 				break;
 			case 'UPDATES':
-			// If the latest item is set then we transfer it to where we want to
+				// If the latest item is set then we transfer it to where we want to
 				if (isset($this->_latest))
```
Comment (elkuku) [inline libraries/joomla/updater/update.php:232]: 5 fails

File: libraries/joomla/user/helper.php [hunk 13]
```diff
@@ -363,28 +363,28 @@ abstract class JUserHelper
...
+						{
+							$new .= $salt;
+						}
+						if ($i % 7)
+						{
+							$new .= $plaintext;
+						}
+						$new .= ($i & 1) ? substr($binary, 0, 16) : $plaintext;
+						$binary = JUserHelper::_bin(md5($new));
 				}
```
Comment (elkuku) [inline libraries/joomla/user/helper.php:388]: this produces 2 tabs instead of one.. ?

File: libraries/joomla/utilities/utility.php [hunk 14]
```diff
@@ -238,7 +238,7 @@ class JUtility
 
 		switch ($last)
 		{
-			// The 'G' modifier is available since PHP 5.1.0
+				// The 'G' modifier is available since PHP 5.1.0
 			case 'g':
```
Comment (elkuku) [inline libraries/joomla/utilities/utility.php:242]: and a final fail

**Comment by elkuku** [comment]
Overall, very nice. Don't trust the auto formatters :P
sorry for the spam :(

File: libraries/joomla/updater/update.php (hunk 12 above)
Comment (realityking) [inline libraries/joomla/updater/update.php:232]: Actually just 4 ;)

**Comment by realityking** [comment]
This should fix all your comments. I probably should fix as much as possible in the autoformatter too, but who has the time ;)

File: libraries/joomla/cache/controller/view.php (hunk 1 above)
Comment (elinw) [inline libraries/joomla/cache/controller/view.php:73]: Comments should definitely be on their own lines above the code they refer to but afaik  the sniffs don't deal with that at all .... but there should no be a tab the { should align with the else.

Comment (elkuku) [inline libraries/joomla/cache/controller/view.php:73]: Should be possible to create such a sniff ;)

Comment (elinw) [inline libraries/joomla/cache/controller/view.php:73]: I have a pretty detailed set of rules that I used in proof reading all of the comments last spring.

E.g. always a space after // 
always start with capital unless it is something that is specifically lower case or a continuation of a sentence.
And so on.

File: libraries/joomla/form/fields/groupedlist.php (hunk 5 above)
Comment (elinw) [inline libraries/joomla/form/fields/groupedlist.php:121]: That comment should not have a tab, it should be aligned with the code it refers to.

File: libraries/joomla/form/rules/url.php (hunk 7 above)
Comment (elinw) [inline libraries/joomla/form/rules/url.php:79]: Definitely should not be a tab there.

**Pull request merged**